packages/analytics-core/src/plugins/destination.ts

```diff
@@ -205,9 +202,10 @@ export class Destination implements DestinationPlugin {
     this.fulfillRequest(list, res.statusCode, SUCCESS_MESSAGE);
   }
 
-  handleInvalidResponse(res: InvalidResponse, list: Context[]) {
+  handleInvalidResponse(res: InvalidResponse, list: Context[], useRetry: boolean) {
     if (res.body.missingField || res.body.error.startsWith(INVALID_API_KEY)) {
-      this.fulfillRequest(list, res.statusCode, res.body.error);
+      this.fulfillRequest(list, res.statusCode, `${res.status}: ${getResponseBodyString(res)}`);
```

Comment:
Does this handle all of the previous cases?
https://github.com/amplitude/Amplitude-TypeScript/pull/568/files#diff-379abfc2a0d72883f73638e5ee6b8a9e0fb89793f5cebc6751a0716731bfa1d7L158-L165

The original code had a special handling for `res` without `res.body` which I think would not be fulfilled with this new logic.

Reply:
`filfillRequest()` triggers callback which can be customized by customers to monitor the SDK errors. So have to make sure we don't miss any.

Since we don't have tests for `filfillRequest()`, its hard to manually check them. 

Recover `filfillRequest()` in send when `!useTry` in https://github.com/amplitude/Amplitude-TypeScript/pull/568/commits/afe973135b0d8a2607d49d5e5a6b3b28cecf28e7. Should not missing any `fulfilllRequest()` not.